types/block_test.go

```diff
@@ -367,6 +368,104 @@ func TestCommitToVoteSet(t *testing.T) {
 	}
 }
 
+func TestCommitToVoteSetWithBadVotes(t *testing.T) {
+	lastID := makeBlockIDRandom()
+	height := int64(3)
+	round := 1
+	numValidators := 10
+
+	voteSet, valSet, vals := randVoteSet(height-1, 1, PrecommitType, numValidators, 1)
+
+	// > 2/3 sign for this block
+	numBlockValidators := numValidators*2/3 + 1
+	for i := 0; i < numBlockValidators; i++ {
+		addr := vals[i].GetPubKey().Address()
+		vote := &Vote{
+			ValidatorAddress: addr,
+			ValidatorIndex:   i,
+			Height:           height - 1,
+			Round:            round,
+			Type:             PrecommitType,
+			BlockID:          lastID,
+			Timestamp:        tmtime.Now(),
+		}
+
+		_, err := signAddVote(vals[i], vote, voteSet)
+		assert.NoError(t, err)
+	}
+
+	// 1/3 vote for a different block
+	for i := numBlockValidators; i < numValidators; i++ {
+		addr := vals[i].GetPubKey().Address()
+		vote := &Vote{
+			ValidatorAddress: addr,
+			ValidatorIndex:   i,
+			Height:           height - 1,
+			Round:            round,
+			Type:             PrecommitType,
+			BlockID:          makeBlockIDRandom(),
+			Timestamp:        tmtime.Now(),
+		}
+
+		_, err := signAddVote(vals[i], vote, voteSet)
+		assert.NoError(t, err)
+	}
+	commit := voteSet.MakeCommit()
+
+	chainID := voteSet.ChainID()
+	voteSet2 := CommitToVoteSet(chainID, commit, valSet)
```

Comment:
Do we have to test CommitToVoteSet func here? Should be enoth to call `MakeCommit` and maybe `commit.ValidateBasic()`

Reply:
I wasn't sure! It does seem like `CommitToVoteSet` does some validation on the vote, so I thought maybe we should test it, too. However, upon closer inspection, it seems like it's not doing any validation on the whole set of votes, so I think you're right.